spring-ai-core/src/main/java/org/springframework/ai/model/function/DefaultFunctionCallbackBuilder.java

```diff
@@ -50,86 +44,11 @@
  * @author Christian Tzolov
  * @since 1.0.0
  */
-public class DefaultFunctionCallbackBuilder implements FunctionCallback.Builder {
+public class DefaultFunctionCallbackBuilder extends DefaultCommonInvokingSpec<FunctionCallback.Builder>
```

Comment:
I've made the DefaultFunctionCallbackBuilder extend DefaultCommonInvokingSpec only for backward compatibility. 
But this can be confusing and might be better to remove it before M5?

Reply:
I vote for removing it before M5 as the current common builder spec is confusing anyway and the users would prefer to switch than stick to it.